xtable-api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java

```diff
@@ -149,19 +151,26 @@ private SyncResult getSyncResult(
       InternalTable tableState,
       SyncFiles fileSyncMethod,
       Instant startTime,
-      List<Instant> pendingCommits) {
+      List<Instant> pendingCommits,
+      String sourceIdentifier) {
     // initialize the sync
     conversionTarget.beginSync(tableState);
+    // Persist the latest commit time in table properties for incremental syncs
```

Comment:
Here We need to move the Metadata set operation earlier because it will be required during the `sync()` operation in Iceberg (Delta and Hudi only need it in `completeSync()`